pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -164,6 +164,20 @@ func (c *Controller) GetService(hostname model.Hostname) (*model.Service, error)
 	return nil, errs
 }
 
+// GetServiceAttributes retrieves the custom attributes of a service if the service exists
+func (c *Controller) GetServiceAttributes(service *model.Service) (*model.ServiceAttributes, error) {
```

Comment:
It's probably better to use hostname as the argument. Only hostname is used anyways.

Reply:
Right, only hostname is used for now but maybe leave some flexibility for other platform?

@rshriram Could you take a look this PR, Thank you!

Reply:
Lets stick to hostname. We can always change ti something else later.